### PR TITLE
JobSite-1101: created signup and login API

### PR DESCRIPTION
**Pull request by 04vignesh45**
Created signup and login API

### CODE REVIEW

File: driver/driver.go [hunk 1]
```diff
@@ -0,0 +1 @@
+package driver
```
Comment (mdufreshers2023) [inline driver/driver.go:1]: Don't commit the empty file

File: handler/handler.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
+package handler
+
+import (
+	"fmt"
```
Comment (mdufreshers2023) [inline handler/handler.go:4]: Need to add comment as build-in package

File: handler/handler.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+	"net/http"
+
```
Comment (mdufreshers2023) [inline handler/handler.go:6]: Need to add comment as user-defined package

File: handler/handler.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+	"echo/models"
+
```
Comment (mdufreshers2023) [inline handler/handler.go:8]: Need to add comment as third party package

File: handler/handler.go [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+	"github.com/dgrijalva/jwt-go"
+	"github.com/labstack/echo/v4"
+)
+
```
Comment (mdufreshers2023) [inline handler/handler.go:12]: What is the purpose of this function? you doesn't called it anywhere

File: handler/handler.go [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+func Signup(c echo.Context) error {
+	var user models.User
+	if err := c.Bind(&user); err != nil {
+		return c.JSON(http.StatusBadRequest, map[string]string{"message": "Invalid request payload"})
+	}
+
+	// Check if the user already exists by ID, Email, or Username
+	for _, u := range models.Users {
+		if u.ID == user.ID || u.Email == user.Email || u.Username == user.Username {
```
Comment (mdufreshers2023) [inline handler/handler.go:53]: Need to separate the errors, Don't give it simply "User already exists"

File: handler/handler.go [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+			return c.JSON(http.StatusUnauthorized, map[string]string{"message": "Invalid token"})
+		}
+
+		if claims, ok := token.Claims.(jwt.MapClaims); ok && token.Valid {
+			userID := claims["id"].(string)
+			for _, user := range models.Users {
+				if user.ID == userID {
+					// Store user information in the context
+					c.Set("user", user)
+					return next(c)
```
Comment (mdufreshers2023) [inline handler/handler.go:36]: What is the purpose of next(c)?

File: handler/handler.go (hunk 6 above)
Comment (mdufreshers2023) [inline handler/handler.go:53]: Need to validate the email, The email must be in correct pattern like .gmail

File: handler/handler.go [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+		}
+		if u.Password != user.Password {
+			return c.JSON(http.StatusUnauthorized, map[string]string{"message": "incorrect password"})
+		}
+
+	}
+	return c.JSON(http.StatusAccepted, map[string]string{"message": "login successful"})
+}
+
+func Welcome(c echo.Context) error {
```
Comment (mdufreshers2023) [inline handler/handler.go:94]: Remove this

File: helper/helper.go [hunk 9]
```diff
@@ -0,0 +1 @@
+package helper
```
Comment (mdufreshers2023) [inline helper/helper.go:1]: Don't upload the empty file

File: main.go [hunk 10]
```diff
@@ -1,7 +1,7 @@
 package main
 
-import "fmt"
+import "echo/router"
 
 func main() {
-	fmt.Println("hello")
+router.Router()
```
Comment (mdufreshers2023) [inline main.go:6]: Need to align it

File: repository/repository.go [hunk 11]
```diff
@@ -0,0 +1 @@
+package repository
```
Comment (mdufreshers2023) [inline repository/repository.go:1]: Don't upload the empty file

File: router/router.go [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+	"github.com/labstack/echo/v4"
+)
+
+func Router() {
+	e := echo.New()
+
+	// Routes
+	e.POST("/signup", handler.Signup)
+	e.POST("/login", handler.Login)
+    e.GET("/",handler.Welcome)
```
Comment (mdufreshers2023) [inline router/router.go:15]: Remove the handler

Comment (mdufreshers2023) [inline router/router.go:15]: Need to align

File: handler/handler.go (hunk 5 above)
Comment (04vignesh45) [reply handler/handler.go:12]: will correct the mistake

**Pull request merged**